botorch/acquisition/multi_step_lookahead.py

```diff
@@ -33,20 +40,23 @@ def __init__(
         model: Model,
         batch_sizes: List[int],
         num_fantasies: List[int],
-        value_function_cls: Type[AcquisitionFunction] = PosteriorMean,
-        value_function_kwargs: Optional[Dict[str, Any]] = None,
+        stage_valfunc_cls: Optional[List[Optional[Type[AcquisitionFunction]]]] = None,
+        stage_valfunc_argfacs: Optional[List[Optional[TAcqfArgConstructor]]] = None,
         objective: Optional[AcquisitionObjective] = None,
-        inner_sampler: Optional[MCSampler] = None,
+        num_inner_mc_samples: Optional[List[int]] = None,
```

Comment:
why not pass in a list of samplers to allow the flexibility of using different samplers?

Reply:
yeah that's a possibility. I was trying to keep the interface somewhat clean to start with, but we could also support both